app/api/utils.py

```diff
@@ -335,6 +345,39 @@ def parse(self, file):
             yield data
 
 
+class ExcelParser(FileParser):
+    def parse(self, file):
+        excel_book = pyexcel.iget_book(file_type="xlsx", file_content=file)
+        data = []
+        # Handle multiple sheets
+        for sheet_name in excel_book.sheet_names():
+            reader = excel_book[sheet_name].to_array()
+            columns = next(reader)
+            # handles header or no header when only text column is given
+            if len(columns) == 1 and columns[0] != 'text':
```

Comment:
At this point the code is mostly the same as the CSV parser. I wonder if the functionality should be shared?

Reply:
Yeah I was thinking the same thing! I think pyexcel can handle csv as well. I'll look into it!

Reply:
Extracted a common function